lib/transition/unpack.c

```diff
@@ -232,6 +357,12 @@ unpack_graph(xmlNode * xml_graph, const char *reference)
         }
     }
 
+    for (sIter = new_graph->synapses; sIter != NULL; sIter = sIter->next) {
+        synapse_t *synapse = (synapse_t *) sIter->data;
+
+        inject_metadata(new_graph, synapse);
```

Comment:
I believe the iterator here is invalidated when inject_metadata() adds to new_graph->synapses. Not sure of best way around that. We don't want to make a copy of the graph. Maybe there's a way to build the synapses to add and add them after the loop.